packages/frontend/src/lib/pnl.ts

```diff
@@ -70,25 +82,47 @@ export async function calcETHCollateralPnl(
           )
         }
         return acc
-      }, Promise.resolve({ deposits: BIG_ZERO, withdrawals: BIG_ZERO }))
-    : { deposits: BIG_ZERO, withdrawals: BIG_ZERO }
+      }, Promise.resolve({ deposits: BIG_ZERO, withdrawals: BIG_ZERO, priceError: false }))
+    : { deposits: BIG_ZERO, withdrawals: BIG_ZERO, priceError: true }
 
-  return currentVaultEthBalance.times(uniswapEthPrice).minus(deposits.minus(withdrawals))
+  return !priceError ? currentVaultEthBalance.times(uniswapEthPrice).minus(deposits.minus(withdrawals)) : BIG_ZERO
+}
+
+const getRelevantSwaps = (squeethAmount: BigNumber, swaps: swaps_swaps[], isWethToken0: boolean, isLong = false) => {
+  let totalSqueeth = BIG_ZERO
+  const relevantSwaps = []
+  for (let index = swaps.length - 1; index >= 0; index--) {
+    const squeethAmt = new BigNumber(isWethToken0 ? swaps[index].amount1 : swaps[index].amount0)
```

Comment:
this var name and the following are pretty confusing imo. would try to use smth that describes what each represents better